api/v1beta1/ibmpowervsmachine_types.go

```diff
@@ -74,6 +74,14 @@ type IBMPowerVSMachineSpec struct {
 	// ProviderID is the unique identifier as specified by the cloud provider.
 	// +optional
 	ProviderID *string `json:"providerID,omitempty"`
+
+	// Region specifies the Power VS Service instance region
+	// +optional
+	Region *string `json:"region,omitempty"`
+
+	// Zone specifies the Power VS Service instance zone
+	// +optional
+	Zone *string `json:"zone,omitempty"`
```

Comment:
keeping them optional will be challenge because is user doesn't provide and then someone set v2 as provider version this will throw an NPE, wondering if we can make these params as required fields.

what other think? @Amulyam24 @Prajyot-Parab

Reply:
@Karthik-K-N if required, lets setup a meeting to discuss this in detail

Reply:
We can get the region and zone using the Power VS service instance id and currently its done like that we are not depending on user to provide it.

Reply:
I think keeping it optional would be fine as the region and zone are always being set before setting the provider ID despite the provider version used.

Reply:
IMO, +1 to move it as part of status, as we have internal logic to deduce their values.